consensus/src/pow/work.rs

```diff
@@ -0,0 +1,111 @@
+#![allow(dead_code)]
+
+use crate::pow::config::Config;
+use crate::pow::helpers::{
+    actual_timespan, check_difficulty, retarget, retarget_block_time, testnet,
+};
+use crate::pow::temp::BlockIndex;
+use crate::pow::Error;
+use common::chain::block::Block;
+use common::primitives::consensus_data::{ConsensusData, PoWData};
+use common::primitives::Idable;
+use common::primitives::{BlockHeight, Compact};
+use common::Uint256;
+
+pub fn check_proof_of_work(hash: Uint256, bits: Compact) -> bool {
+    if let Ok(target) = Uint256::try_from(bits) {
+        hash > target
+    } else {
+        false
+    }
+}
+
+impl Config {
+    pub fn check_for_work_required(
+        &self,
+        prev_block_index: &BlockIndex,
+        _height: BlockHeight,
+    ) -> Result<Compact, Error> {
+        //TODO: check prev_block_index exists
+
+        // TODO: only for testnet
+        // if check_difficulty_interval(height) {
+        //     if let ChainType::Testnet = chain_type {
+        //         return Ok(self.next_work_required_for_testnet(time, prev_block));
+        //     }
+        // }
+
+        let retarget_block_time = retarget_block_time(prev_block_index);
+        self.next_work_required(retarget_block_time, prev_block_index)
+    }
+
+    /// retargeting proof of work
+    fn next_work_required(
+        &self,
+        retarget_block_time: u32,
+        prev_block_index: &BlockIndex,
+    ) -> Result<Compact, Error> {
+        let pow_limit = self.limit;
+        let prev_block_bits = prev_block_index.data.bits();
+
+        if self.no_retargeting {
+            return Ok(prev_block_bits);
+        }
+
+        // limit adjustment step
+        let actual_timespan_of_last_2016_blocks =
+            actual_timespan(prev_block_index.get_block_time(), retarget_block_time);
+
+        // retarget
+        retarget(
+            actual_timespan_of_last_2016_blocks,
+            prev_block_bits,
+            pow_limit,
+        )
+    }
+
+    fn next_work_required_for_testnet(&self, time: u32, prev_block_index: &BlockIndex) -> Compact {
+        let pow_limit = Compact::from(self.limit);
+
+        if self.allow_min_difficulty_blocks {
```

Comment:
So why do we have a separate function for testnet if we do this check which is true anyway for testnet? Please combine both functions into one function and make them depend on ChainConfig.